File: src/main/java/ml/wonwoo/springdatashowcase/support/EntityCallbackConfiguration.java
package ml.wonwoo.springdatashowcase.support;

import ml.wonwoo.springdatashowcase.event.Event;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.data.mongodb.core.mapping.event.BeforeConvertCallback;
import org.springframework.data.mongodb.core.mapping.event.BeforeSaveCallback;

@Configuration
public class EntityCallbackConfiguration {

    @Bean
    BeforeSaveCallback<Event> beforeSaveCallback() {
        return (entity, document, collection) -> entity;
    }

    @Bean
    BeforeConvertCallback<Event> beforeConvertCallback() {
        return (entity, collection) -> entity;
    }
}

File: kotlin/src/main/kotlin/ml/wonwoo/springdatashowcase/support/EntityCallbackConfiguration.kt
package ml.wonwoo.springdatashowcase.support

import ml.wonwoo.springdatashowcase.event.Event
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.data.mongodb.core.mapping.event.BeforeConvertCallback
import org.springframework.data.mongodb.core.mapping.event.BeforeSaveCallback

@Configuration(proxyBeanMethods = false)
class EntityCallbackConfiguration {

    @Bean
    fun beforeSaveCallback(): BeforeSaveCallback<Event> {

        return BeforeSaveCallback { entity, _, _ -> entity }

    }

    @Bean
    fun beforeConvertCallback(): BeforeConvertCallback<Event> {

        return BeforeConvertCallback { entity, _ -> entity }

    }

}